eng/scripts/splittestdependencies/Generate-Dependency-functions.ps1

```diff
@@ -0,0 +1,72 @@
+# Take the input of the project list file. Read all files out and split them into groups, and return the group array. 
+# Give the option of whether exclude the target service project files or not.
+
+function Split-Project-File-To-Groups($ProjectFile, $NumberOfTestsPerJob, $ExcludeService, $ServiceDir) {
+    $projects = Get-Content $ProjectFile
+    if ($ExcludeService) {
+      $projectsAfterExclusion = @()
+      foreach ($pkg in $projects) {
+        if ($pkg -notmatch "sdk[\\/]+${ServiceDir}[\\/]+") {
+          $projectsAfterExclusion += $pkg
+        }
+      }
+      $projects = $projectsAfterExclusion
+    }
+    $numOfGroups = [Math]::Floor([decimal]($projects.Count / $NumberOfTestsPerJob ))
+    $projGroup = [System.Collections.ArrayList]::new()
+    for ($i = 0; $i -lt ($numOfGroups - 1); $i++) {
+      $projGroup.Add($projects[( $i * $NumberOfTestsPerJob )..(($i + 1) * $NumberOfTestsPerJob - 1)]) > $null
+    }
+    if ($projects.Count -gt ($numOfGroups - 1) * $NumberOfTestsPerJob) {
+      $projGroup.Add($projects[(($numOfGroups - 1) * $NumberOfTestsPerJob)..($projects.Count - 1)]) > $null
+    }
+    return $projGroup
+}
+
+
+# Take the group contains the project list. Write each group into a project file which can be used by dotnet test command.
+# Return the array of project files
+function Write-Test-Dependency-Group-To-Files($ProjectGroups, $MatrixOutputFolder) {
+    $null = New-Item -Path $MatrixOutputFolder -ItemType "directory" -Force 
+    $numOfGroups = $ProjectGroups.Count
+    $allOutputFiles = @()
+    for ($i = 0; $i -lt $numOfGroups; $i++) {
+      $projectFilePath = "Project_$i.props"
+      $null = New-Item -Path "$MatrixOutputFolder/$projectFilePath" -ItemType "file" -Force
+      # Retain the structure without ProjectReference.
+      $templateXml=[xml]'<Project></Project>'  
+      $projectNode = $templateXml.SelectNodes("/Project")
+      $itemGroupNode=$templateXml.CreateNode("element", "ItemGroup", "")
+      $projectNode.AppendChild($itemGroupNode) > $null
+      foreach($pkg in $ProjectGroups[$i]) {
+        $newElem = $templateXml.CreateNode("element", "ProjectReference", "")
+        $newElemAttr = $templateXml.CreateAttribute("Include")
+        $newElemAttr.InnerText = $pkg
+        $newElem.Attributes.Append($newElemAttr) > $null
+        
+        $itemGroupNode.AppendChild($newElem) > $null
+        $null = $templateXml.Save("azure-sdk-for-net/$MatrixOutputFolder/$projectFilePath")
```

Comment:
I think this should be outside this inner for loop? Otherwise you are incrementally updating the file multiple times vs. a single update after you are done building `$templateXml`?

Reply:
Also why does `azure-sdk-for-net` need to be hardcoded here?

Reply:
My local is doing some weird stuff. It saves to somewhere outside of `azure-sdk-for-net`

Reply:
Ideally the location is predictable based on $PSScriptRoot or something?